uh50_api/uh50.py

```diff
@@ -2,70 +2,300 @@
+        if match:
+            try:
+                self.fault_hours_previous_year = int(match.group(1))
+            except:
+                self.fault_hours_previous_year = match.group(1)
+
+        # yearly_set_day
+        match = re.search(UH50_REGEX_CONFIG['yearly_set_day'], str(data), re.M|re.I)
+        if match:
+            self.yearly_set_day = match.group(1)
+
+        # monthly_set_day
+        match = re.search(UH50_REGEX_CONFIG['monthly_set_day'], str(data), re.M|re.I)
+        if match:
+            self.monthly_set_day = match.group(1)
+
+        # meter_date_time
+        match = re.search(UH50_REGEX_CONFIG['meter_date_time'], str(data), re.M|re.I)
+        if match:
+            try:
+                self.meter_date_time = datetime.strptime(match.group(1),'%Y-%m-%d&%H:%M:%S')
+            except:
+                self.meter_date_time = match.group(1)
+
+        # measuring_range_m3ph
+        match = re.search(UH50_REGEX_CONFIG['measuring_range_m3ph'], str(data), re.M|re.I)
+        if match:
+            try:
+                self.measuring_range_m3ph = float(match.group(1))
+            except:
+                self.measuring_range_m3ph = match.group(1)
+
+        # settings_and_firmware
+        match = re.search(UH50_REGEX_CONFIG['settings_and_firmware'], str(data), re.M|re.I)
+        if match:
+            self.settings_and_firmware = match.group(1)
+
+        # flow_hours
+        match = re.search(UH50_REGEX_CONFIG['flow_hours'], str(data), re.M|re.I)
+        if match:
+            try:
+                self.flow_hours = int(match.group(1))
+            except:
+                self.flow_hours = match.group(1)
+
+        return
+ 
+    def validate(self) -> Serial:
+        try:
+            result = self._connect_serial()
+            result.close
+        except:
+            raise
+
+    def _connect_serial(self) -> Serial:
+        "Make the connection to the serial device"
+        return serial.Serial(self.port,
+                            baudrate=300,
+                            bytesize=serial.SEVENBITS,
+                            parity=serial.PARITY_EVEN,
+                            stopbits=serial.STOPBITS_TWO,
+                            timeout=1,
+                            xonxoff=0,
+                            rtscts=0
+                            )
+
+    def _wake_up(self, conn) -> str:
+        "Waking up the device and get the model number"
+        # Waking up should be done at 300 baud
+        # Sending /?!
+        conn.write(b"\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x2F\x3F\x21\x0D\x0A")
+        ir_command='/?!\x0D\x0A'
+        conn.write(ir_command.encode('utf-8'))
+        conn.flush()
+        return conn.readline().decode('utf-8')[1:9]   # Read at 300 baud, this gives us the typenr
+
+    def _get_data(self, conn):
```

Comment:
in _get_data staan wat magic numbers. Baudrate lijkt me prima config (lees constructor arg) voor de uh50 class.

Reply:
Ik zie dat je de ene keer 300 en de andere keer 2400 nodig hebt om te lezen. Dat zouden dan 2 constructor args worden.

Reply:
Dit heb ik iets anders gedaan. In volgende PR even zien of dat oke is.